### PR TITLE
Add validation for a few image_desc_t members.

### PR DESCRIPTION
**Pull request by aarongreig**
Also removes these checks from the cuda and hip adapters since they should be caught by the validation layer now.

The new validation rules bring image_desc_t in line with the CL struct it was derived from. We already had CTS coverage for them.

LLVM testing https://github.com/intel/llvm/pull/12267

### CODE REVIEW

**Comment by aarongreig** [comment]
I want to remove the "must be zero" members from the struct, still looking into their original purpose so I can write up a proper issue about it.

**Review by pbalcer — APPROVED** [review]

**Comment by fabiomestre** [comment]
I have updated the target branch of this PR from the `adapters` branch to the `main` branch.
Development in UR is moving back to `main`. The `adapters` branch will soon be deleted.

**Comment by aarongreig** [comment]
@oneapi-src/unified-runtime-cuda-write @oneapi-src/unified-runtime-hip-write the changes here are trivial but I don't really want to establish a precedent for merging stuff without code owner approval

**Review by hdelan — APPROVED** [review]

**Comment by aarongreig** [comment]
absorbed into https://github.com/oneapi-src/unified-runtime/pull/1212

**Pull request closed without merging**
